radar-upload-backend/src/main/java/org/radarbase/upload/dto/RecordMapperImpl.kt

```diff
@@ -46,7 +44,27 @@ class RecordMapperImpl: RecordMapper {
             modifiedDate = metadata.modifiedDate,
             committedDate = metadata.committedDate,
             logs = metadata.logs?.let {
-                LogsDto(url = "${uri.baseUri}/records/${metadata.id}/logs")
+                LogsDto(url = "${uri.baseUri}/records/${metadata.record.id}/logs")
             }
     )
+
+    override fun toMetadata(metadata: RecordMetadataDTO, origin: RecordMetadata) = RecordMetadata()
+            .also {
+                it.revision = metadata.revision
+                it.message = metadata.message
+                it.status = RecordStatus.valueOf(metadata.status)
+                if(origin != null) {
+                    it.createdDate = origin.createdDate
+                    it.committedDate = origin.committedDate
+                    it.modifiedDate = origin.modifiedDate
+                    it.logs = origin.logs
+                }
+            }
+
+    override fun fromMetadataToLogs(metadata: RecordMetadata) : LogsDto {
+        // read the first 5000 char from Clob
+        return LogsDto(
+                contents = metadata.logs?.logs?.getSubString(0, 5000),
+                url ="${uri.baseUri}/records/${metadata.record.id}/logs")
```

Comment:
you can save a fetch by excluding using `metadata.id` instead of `metadata.record.id`.